stringdtype/stringdtype/src/umath.c

```diff
@@ -13,6 +13,65 @@
 #include "string.h"
 #include "umath.h"
 
+static int
+add_resolve_descriptors(struct PyArrayMethodObject_tag *NPY_UNUSED(method),
+                        PyArray_DTypeMeta *NPY_UNUSED(dtypes[]),
+                        PyArray_Descr *given_descrs[],
+                        PyArray_Descr *loop_descrs[],
+                        npy_intp *NPY_UNUSED(view_offset))
+{
+    Py_INCREF(given_descrs[0]);
+    loop_descrs[0] = given_descrs[0];
+    Py_INCREF(given_descrs[1]);
+    loop_descrs[1] = given_descrs[1];
+
+    StringDTypeObject *new = new_stringdtype_instance();
+    if (new == NULL) {
+        return -1;
```

Comment:
The design is currently to ask you to clean up here.  So its to put it this first (IMO).

But you should not need a resolve descriptors at all, since the dtype isn't parametric?

Reply:
You're totally right, I moved this to the top.

After reading more, I modified `init_ufunc` to accept `NULL` as a way of indicating that the ufunc uses `default_resolve_descriptors`. I was able to eliminate another place where this was happening as well for the `min`/`max` ufuncs.